system-tests/services/test-apps.js

```diff
@@ -1677,5 +1677,186 @@ describe("Services", function() {
         .getFormGroupInputFor("Container Path")
         .should("have.value", "test");
     });
+
+    it("should create an app with virtual network", function() {
+      const serviceName = "app-with-virtual-network";
+      const command = "python3 -m http.server 8080";
+      const containerImage = "python:3";
+
+      cy.contains("Single Container").click();
+
+      cy
+        .root()
+        .getFormGroupInputFor("Service ID *")
+        .type(`{selectall}{rightarrow}${serviceName}`);
+
+      // Hack to allow entering decimals in number fields
+      cy.root().getFormGroupInputFor("CPUs *").invoke("attr", "type", "text");
+      cy.root().getFormGroupInputFor("CPUs *").type("{selectall}0.5");
+      cy.root().getFormGroupInputFor("CPUs *").invoke("attr", "type", "number");
+
+      cy.root().getFormGroupInputFor("Memory (MiB) *").type("{selectall}32");
+      cy.root().getFormGroupInputFor("Container Image").type(containerImage);
+      cy.root().getFormGroupInputFor("Command").type(command);
+
+      cy.get(".menu-tabbed-item").contains("Networking").click();
+
+      cy
+        .root()
+        .getFormGroupInputFor("Network Type")
+        .select("Virtual Network: dcos");
+
+      cy.get(".button").contains("Add Service Endpoint").click();
+      cy.get('input[name="portDefinitions.0.containerPort"]').type("8080");
+      cy.get('input[name="portDefinitions.0.name"]').type("http");
+
+      cy.get(".button").contains("Add Service Endpoint").click();
+      cy.get('input[name="portDefinitions.1.containerPort"]').type("8080");
+      cy.get('input[name="portDefinitions.1.name"]').type("mapped");
+
+      cy
+        .get('input[name="portDefinitions.1.portMapping"]')
+        .parents(".form-control-toggle")
+        .click();
+
+      cy
+        .get('input[name="portDefinitions.1.automaticPort"]')
+        .parents(".form-control-toggle")
+        .click();
+
+      cy.root().getFormGroupInputFor("Host Port").type("4200");
+
+      cy.get("label").contains("JSON Editor").click();
+
+      cy.get("#brace-editor").contents().asJson().should("deep.equal", [
+        {
+          id: `/${Cypress.env("TEST_UUID")}/${serviceName}`,
+          cmd: command,
+          cpus: 0.5,
+          mem: 32,
+          instances: 1,
+          container: {
+            type: "DOCKER",
+            docker: {
+              image: containerImage,
+              network: "USER",
+              portMappings: [
+                {
+                  name: "http",
+                  containerPort: 8080
+                },
+                {
+                  name: "mapped",
+                  hostPort: 4200,
+                  containerPort: 8080,
+                  protocol: "tcp"
+                }
+              ]
+            }
+          },
+          ipAddress: {
```

Comment:
This looks like the old network configuration. Perhaps you should migrate this to the new version.
Have a look here: https://github.com/mesosphere/marathon/blob/master/docs/docs/upgrade/network-api-migration.md

Even if the tests are passing now, the will definitely fail when we make the new release 😄